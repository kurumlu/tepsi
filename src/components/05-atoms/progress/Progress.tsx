import * as React from 'react';
import { FunctionComponent } from 'react';

export type ProgressProps = {
  progress: number;
};

export const Progress: FunctionComponent<ProgressProps> = ({ progress }) => (
  <div className="as-a-progress">
    <div
      className="as-a-progress__bar"
      style={{
        width: `${progress > 100 ? 100 : progress < 0 ? 0 : progress}%`,
      }}></div>
  </div>
);
