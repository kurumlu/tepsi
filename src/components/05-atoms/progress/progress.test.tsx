import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Progress, ProgressProps } from './Progress';

describe('<Progress />', () => {
  it('should render with default props', () => {
    const wrapper: ShallowWrapper<ProgressProps> = shallow(
      <Progress progress={25} />
    );

    expect(wrapper).toMatchSnapshot('default');
  });

  it('should set 100% progress if passed value is greater than 100', () => {
    const wrapper: ShallowWrapper<ProgressProps> = shallow(
      <Progress progress={110} />
    );

    expect(wrapper.find('.sd-a-progress__bar').props().style.width).toBe(
      '100%'
    );
  });

  it('should set 0% progress if passed value is less than 0', () => {
    const wrapper: ShallowWrapper<ProgressProps> = shallow(
      <Progress progress={-10} />
    );

    expect(wrapper.find('.sd-a-progress__bar').props().style.width).toBe('0%');
  });
});
