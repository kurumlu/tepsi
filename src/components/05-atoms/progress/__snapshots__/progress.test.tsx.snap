// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> should render with default props: default 1`] = `
<div
  className="as-a-progress"
>
  <div
    className="as-a-progress__bar"
    style={
      Object {
        "width": "25%",
      }
    }
  />
</div>
`;
